test/services/twilio-api/twilio-client.test.js

```diff
@@ -265,6 +265,230 @@ describe('services', () => {
+              { accountSid, httpClient, edge: 'edge' }
+            );
+
+            const response = await client.create({
+              domain: 'api',
+              path: '/2010-04-01/Accounts/{AccountSid}/Messages.json'
+            });
+            expect(response).to.eql({ status: 'queued' });
+          });
+
+        defaultRegionTest
+          .it('uses the default region if edge is provided', async () => {
+            const client = new TwilioApiClient(
+              constants.FAKE_API_KEY, constants.FAKE_API_SECRET,
+              { accountSid, httpClient }
+            );
+
+            const response = await client.create({
+              domain: 'api',
+              path: '/2010-04-01/Accounts/{AccountSid}/Messages.json',
+              edge: 'edge'
+            });
+            expect(response).to.eql({ status: 'queued' });
+          });
+
+        const regionTest = test.nock('https://api.region.twilio.com', api => {
+          api.post(`/2010-04-01/Accounts/${accountSid}/Messages.json`).reply(201, {
+            status: 'queued'
+          });
+        });
+
+        regionTest.it('uses the client region if defined', async () => {
+          const client = new TwilioApiClient(
+            constants.FAKE_API_KEY, constants.FAKE_API_SECRET,
+            { accountSid, httpClient, region: 'region' }
+          );
+
+          const response = await client.create({
+            domain: 'api',
+            path: '/2010-04-01/Accounts/{AccountSid}/Messages.json',
+            // Should ignore the region in the uri
+            uri: 'https://api.uriRegion.twilio.com'
+          });
+          expect(response).to.eql({ status: 'queued' });
+        });
+
+        regionTest.it('uses the provided region if client region defined and region is provided', async () => {
+          const client = new TwilioApiClient(
+            constants.FAKE_API_KEY, constants.FAKE_API_SECRET,
+            { accountSid, httpClient, region: 'region2' }
+          );
+
+          const response = await client.create({
+            domain: 'api',
+            path: '/2010-04-01/Accounts/{AccountSid}/Messages.json',
+            region: 'region',
+            // Should ignore the region in the uri
+            uri: 'https://api.uriRegion.twilio.com'
+          });
+          expect(response).to.eql({ status: 'queued' });
+        });
+
+        regionTest.it('uses the provided region', async () => {
+          const client = new TwilioApiClient(
+            constants.FAKE_API_KEY, constants.FAKE_API_SECRET,
+            { accountSid, httpClient }
+          );
+
+          const response = await client.create({
+            domain: 'api',
+            path: '/2010-04-01/Accounts/{AccountSid}/Messages.json',
+            region: 'region',
+            // Should ignore the region in the uri
+            uri: 'https://api.uriRegion.twilio.com'
+          });
+          expect(response).to.eql({ status: 'queued' });
+        });
```

Comment:
Looks just like the previous test.

Reply:
The previous test has the `client.region`, `opts.region` and uri region all defined, and asserts that `opts.region` takes precedence. This test does not have `client.region` defined and asserts it uses `opts.region`